### PR TITLE
Add Basic auth tests and fix Windows auth tests

### PR DESCRIPTION
**Pull request by pakrym**
https://github.com/aspnet/IISIntegration/issues/1429

### CODE REVIEW

File: test/IISExpress.FunctionalTests/BasicAuthTests.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                .WithTfms(Tfm.NetCoreApp22)
+                .WithApplicationTypes(ApplicationType.Portable)
+                .WithAllAncmVersions()
+                .WithAllHostingModels();
+
+        [ConditionalTheory(Skip = "This test is manual. To run it set username and password variables below")]
+        [MemberData(nameof(TestVariants))]
+        public async Task BasicAuthTest(TestVariant variant)
+        {
+            var username = "SET THIS VARIABLE";
```
[the remark this answers is not in the document]
Comment (jkotalik) [reply test/IISExpress.FunctionalTests/BasicAuthTests.cs]: Maybe force the user to send the http request. Test is already semi manual.

[the remark this answers is not in the document]
Comment (jkotalik) [reply test/IISExpress.FunctionalTests/BasicAuthTests.cs]: Or an environment variable/git ignored file.

**Comment by jkotalik** [comment]
There are some out of proc tests you will need to fix: Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.HelloWorldTests.HelloWorld

File: test/IISExpress.FunctionalTests/BasicAuthTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tratcher) [reply test/IISExpress.FunctionalTests/BasicAuthTests.cs]: Environment variable.

**Comment by pakrym** [comment]
:up::date:

**Pull request merged**